File: packages/python/staffkm-sdk/staffkm/resources/billing.py
"""Billing resource — admin scope."""
from __future__ import annotations

import httpx


class _Users:
    def __init__(self, http: httpx.Client):
        self._http = http

    def list(self, month: str | None = None) -> list[dict]:
        params = {"month": month} if month else {}
        r = self._http.get("/api/v1/billing/users", params=params)
        r.raise_for_status()
        return r.json().get("data") or []

    def detail(self, user_id: str, month: str | None = None) -> dict:
        params = {"month": month} if month else {}
        r = self._http.get(f"/api/v1/billing/users/{user_id}", params=params)
        r.raise_for_status()
        return r.json().get("data") or r.json()

    def csv(self, month: str | None = None) -> bytes:
        params = {"month": month} if month else {}
        r = self._http.get("/api/v1/billing/users.csv", params=params)
        r.raise_for_status()
        return r.content


class BillingResource:
    def __init__(self, http: httpx.Client):
        self.users = _Users(http)

File: packages/ts/staffkm-sdk/src/resources/billing.ts
import { AxiosInstance } from 'axios'

class BillingUsers {
  constructor(private http: AxiosInstance) {}

  async list(month?: string): Promise<any[]> {
    const r = await this.http.get('/api/v1/billing/users', { params: month ? { month } : {} })
    return r.data?.data ?? []
  }

  async detail(userId: string, month?: string): Promise<any> {
    const r = await this.http.get(`/api/v1/billing/users/${userId}`, {
      params: month ? { month } : {},
    })
    return r.data?.data ?? r.data
  }

  async csv(month?: string): Promise<string> {
    const r = await this.http.get('/api/v1/billing/users.csv', {
      params: month ? { month } : {},
      responseType: 'text',
    })
    return r.data as string
  }
}

export class BillingResource {
  users: BillingUsers
  constructor(http: AxiosInstance) {
    this.users = new BillingUsers(http)
  }
}
